core/src/components/datetime/utils/format.ts

```diff
@@ -49,16 +49,26 @@ export const addTimePadding = (value: number): string => {
 };
 
 /**
- * Formats the hour value so that it
- * is always 2 digits. Only applies
- * if using 12 hour format.
+ * Formats 24 hour times so that
+ * it always has 2 digits. For
+ * 12 hour times it ensures that
+ * hour 0 is formatted as '12'.
  */
 export const getFormattedHour = (hour: number, use24Hour: boolean): string => {
-  if (!use24Hour) {
-    return hour.toString();
+  if (use24Hour) {
```

Comment:
I flipped the order around so that we check positive cases first (IMO I think the function is easier to understand as a result)